slack-java-client/src/main/java/com/hubspot/slack/client/SlackWebClient.java

```diff
@@ -716,6 +718,45 @@ protected Long getInitialOffset() {
     return postSlackCommand(SlackMethods.conversations_open, params, ConversationsOpenResponse.class);
   }
 
+  @Override
+  public Iterable<CompletableFuture<Result<List<String>, SlackError>>> getConversationMembers(ConversationMemberParams params) {
+    return new AbstractPagedIterable<Result<List<String>, SlackError>, String>() {
+      @Override
+      protected String getInitialOffset() {
+        return null;
+      }
+
+      @Override
+      protected LazyLoadingPage<Result<List<String>, SlackError>, String> getPage(String offset) throws Exception {
+        if (LOG.isTraceEnabled()) {
+          LOG.trace("Fetching slack usergroup page from {}", offset);
+        }
+
+        CompletableFuture<Result<ConversationMemberResponse, SlackError>> resultFuture = postSlackCommand(
+                SlackMethods.conversations_members,
+                ConversationMemberParams.builder()
+                        .from(params)
+                        .build(),
```

Comment:
We have a pattern for paged APIs where we configure the default batch size (`limit`) in the `SlackClientRuntimeConfig`. 

https://github.com/HubSpot/slack-client/blob/25232a17884708c860880b917e8dafaf89fb6ea7/slack-java-client/src/main/java/com/hubspot/slack/client/SlackClientRuntimeConfigIF.java#L27-L30


Right now, none of our Paged methods utilize the caller's `limit`. I think it makes sense to take the user passed in `limit` if it exists. If it doesn't exist, we should fall back to the default `SlackClientRuntimeConfig` value

e.g.
```java
ConversationMemberParams.builder builder = ConversationMemberParams.builder()
                        .from(params);
if (!params. getLimit().isPresent()) {
    builder.setLimit(config.getConversationsMembersBatchSize().get())
}
...
```

Reply:
Agreed. Plus Slack has really been pushing for 1000 limit pages in many of their API calls, so I can see them rolling that out for everything.

Reply:
When I write this I focused on method "listUsergroups" that have same realisation (same mistakes: no offset and no batch size)
Now, I try to look difference in slack docs and don't find information about paging in [usergroup_list
](https://api.slack.com/methods/usergroups.list)

Reply:
Yeah, there are some inconsistencies in the APIs unfortunately